user/user.go

```diff
@@ -348,3 +348,67 @@ func GetExecUser(userSpec string, defaults *ExecUser, passwd, group io.Reader) (
 
 	return user, nil
 }
+
+// GetAdditionalGroupsPath looks up a list of groups by name or group id
+// against the group file. If a group name cannot be found, an error will be
+// returned. If a group id cannot be found, it will be returned as-is.
+func GetAdditionalGroupsPath(additionalGroups []string, groupPath string) (gids []int, err error) {
+	groupReader, err := os.Open(groupPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open group file: %v", err)
+	}
+	defer groupReader.Close()
+
+	groups, err := ParseGroupFilter(groupReader, func(g Group) bool {
+		var found bool
+		for _, ag := range additionalGroups {
+			if g.Name == ag || strconv.Itoa(g.Gid) == ag {
+				found = true
+				break
+			}
+		}
+		return found
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to find additional groups %v: %v", additionalGroups, err)
+	}
+
+	gidMap := make(map[string]int)
+	// if we found any group entries that matched our filter, let's take the
+	// first one as "correct"
+	for _, g := range groups {
+		if _, ok := gidMap[g.Name]; !ok {
+			gidMap[g.Name] = g.Gid
+		}
+	}
+	// check list of provided groups again for any other numeric groups which
+	// are not present in the group file
+	for _, ag := range additionalGroups {
+		if _, ok := gidMap[ag]; !ok {
+			addGroup, err := strconv.Atoi(ag)
+			if err != nil {
+				return nil, fmt.Errorf("Unable to find group %s", ag)
+			}
+			// Ensure gid is inside gid range.
+			if addGroup < minId || addGroup > maxId {
+				return nil, ErrRange
+			}
+			// only add the group if we dont have it in the gid map values
+			var found bool
+			for _, gid := range gidMap {
+				if addGroup == gid {
+					found = true
+					break
+				}
+			}
+			if !found {
```

Comment:
Here found can't be true, because you're break from loop when set true. So you don't need this variable at all.